docs/feature/search/vector.md

```diff
@@ -0,0 +1,42 @@
+(vector-search)=
+# Vector Search
+
+:::{rubric} Overview
+:::
+Vector search is a method in artificial intelligence and data retrieval that
+uses mathematical vectors to represent and efficiently search through complex,
+unstructured data.
+
+:::{rubric} About
+:::
+Vector search leverages machine learning (ML) to capture the meaning and
+context of unstructured data, including text and images, transforming it
+into a numeric representation.
+
+Frequently used for semantic search, vector search finds similar data using
+approximate nearest neighbor (ANN) algorithms. Compared to traditional keyword
+search, vector search yields more relevant results and executes faster.
+
+:::{rubric} Details
+:::
+CrateDB can be used as a [vector database] (VDBMS) for storing and retrieving
+vector embeddings based on the [FLOAT_VECTOR] data type and its accompanying 
+[KNN_MATCH] function, effectively conducting HNSW semantic similarity searches
+on them, also known as vector search.
+
+
+## Learn
+
+To learn more about vector search, please visit the corresponding page about
+[](#vector-store).
+
+
+:::{todo}
+Bring page into the same shape like the others in this section.
+Maybe just move the [](#vector-store) page here without further ado.
+:::
```

Comment:
Better repurpose and dissolve the main-level [Vector Store](https://cratedb.com/docs/guide/feature/vector/) page completely, and just move it here, in order to provide a better narrative and enhanced guidance. wdyt?

Reply:
What do you think about this, @surister? In my impression, looking at it from a fresh perspective, listing "Vector Store" as a main feature does not make too much sense, because you will mostly use it for "SEARCHing".

In this spirit, I am planning to move that page to the revamped /search/ section, in order to make the trio of (fts, vector, hybrid) complete. Do you agree?

Reply:
Hm. Thinking about it once more, the Vector Store page might still make sense to have "Vector" enumerated on the top menu level in that section. Otherwise, it might make people feel it is missing here.

Maybe still keep that page, but switch the content around?

Reply:
Hi again. I decided to [refactor mercilessly](http://www.extremeprogramming.org/rules/refactor.html), and dissolved both the "Vector Store" and the "Geospatial Data" page, refactoring them into "Vector Search" and "Geospatial Search", now making up a quartett together with the newly added "Hybrid Search" page.

![image](https://github.com/user-attachments/assets/fef15006-db6e-4f8c-b324-05f4aad04773)

Reply:
That other patch adds relevant redirects to compensate for refactored pages now available on a different location.

- https://github.com/crate/infrastructure/pull/3197